sm/sm.h

```diff
@@ -709,8 +709,18 @@ M0_INTERNAL bool m0_sm_timeout_is_armed(const struct m0_sm_timeout *to);
 /**
  * Posts an AST to a group.
  *
- * An AST must not be re-posted until its previous (already posted) execution
- * completes.
+ * ASTs can be posted many times and concurrently by several threads, the
+ * callbacks will be executed sequentially and in order under the group
```

Comment:
How can this be true? If an ast is posted many times, before the previous instance is removed from the queue, the call-back will be only executed once. This should be clarified in the documentation.

Reply:
Agree, it's a bit confusing.

I meant that after the callback is taken out of the `s_forkq` already at `m0_sm_asts_run()` but not executed yet and posted again - it will be added into the `s_forkq` again and executed again later. But it's too many details for the user.

Updated, please check it now.